File: common_libs/src/main/java/io/virgo_common/common_libs/customView/shimmer/ShimmerViewBase.java
package io.virgo_common.common_libs.customView.shimmer;

/**
 * Shimmer
 * User: romainpiel
 * Date: 10/03/2014
 * Time: 17:33
 */
public interface ShimmerViewBase {

    float getGradientX();
    void setGradientX(float gradientX);
    boolean isShimmering();
    void setShimmering(boolean isShimmering);
    boolean isSetUp();
    void setAnimationSetupCallback(ShimmerViewHelper.AnimationSetupCallback callback);
    int getPrimaryColor();
    void setPrimaryColor(int primaryColor);
    int getReflectionColor();
    void setReflectionColor(int reflectionColor);
}

File: common_libs/src/main/java/io/virgo_common/common_libs/customView/shimmerK/ShimmerViewBase.kt
package io.virgo_common.common_libs.customView.shimmerK

import io.virgo_common.common_libs.customView.shimmerK.ShimmerViewHelper.AnimationSetupCallback

/**
 * Shimmer
 * User: romainpiel
 * Date: 10/03/2014
 * Time: 17:33
 *
 * Source: `https://github.com/romainpiel/Shimmer-android`
 */
interface ShimmerViewBase {
	var gradientX: Float
	var isShimmering: Boolean
	val isSetUp: Boolean
	fun setAnimationSetupCallback(callback: AnimationSetupCallback?)
	var primaryColor: Int
	var reflectionColor: Int
}
